src/entities/communitySbt.ts

```diff
@@ -174,6 +205,166 @@ class SBT {
+                    } else if (transaction.methodId === multiRedeemId) {
+                        const badgeTypes = this.decodeMultipleBadgeTypes(transaction.input);
+                        for (const badgeType of badgeTypes) {
+                            txBadges.set(badgeType, status);
+                        }
+                    }
+                    
+                }
+            }
+
+            // if badges of interest are not part of those 50 transactions, set them as pending
+            for (const badgeType of args.potenetialClaimingBadges) {
+                if (!txBadges.get(badgeType)) {
+                    txBadges.set(badgeType, TxBadgeStatus.PENDING);
+                }
+            }
+
+            // badges claiming status in subgraph - includes all bades earned by user in given season
+            const subgraphClaimedBadges = await this.claimedBadgesInSubgraph(args.subgraphUrl, userAddress, args.season);
+
+            // final claiming status verdict
+            const badgeStatus = subgraphClaimedBadges.map((badge) => {
+                if(badge.claimingStatus === BadgeClaimingStatus.CLAIMED){
+                    return badge;
+                } else {
+                    const txStatus = txBadges.get(badge.badgeType);
+
+                    // badge not found in recent successful txs or in potential pending txs
+                    // meaning their status is desided by the subgraph
+                    if (!txStatus || txStatus === TxBadgeStatus.FAILED) { 
+                        return {
+                            badgeType: badge.badgeType, 
+                            claimingStatus: badge.claimingStatus
+                        }
+                    } else { // subgraph is not updated yet
+                        return {
+                            badgeType: badge.badgeType, 
+                            claimingStatus: BadgeClaimingStatus.CLAIMING
+                        }
+                    }
+                }
+            })
+
+            return badgeStatus;
+        } else {
+            throw new Error("No provider found")
+        }
+        
+    }
+
+    public decodeBadgeType(input: Bytes): number {
+        const inter = new ethers.utils.Interface(CommunitySBT__factory.abi);
+        const decoded = inter.decodeFunctionData("redeem", input);
+        console.log(decoded[0])
+        const metadataURI = decoded[0].metadataURI;
+        const filenamme = metadataURI.split('/')[3];
+        const badgeType = parseInt(filenamme.split('.')[0]);
+
+        return badgeType;
+    }
+
+    public decodeMultipleBadgeTypes(input: Bytes): number[] {
+        let badgeTypes = new Array<number>;
+        const inter = new ethers.utils.Interface(CommunitySBT__factory.abi);
+        const decoded = inter.decodeFunctionData("multiRedeem", input);
+        for (const leafInfo of decoded[0]) {
+            console.log(leafInfo);
+            const metadataURI = leafInfo.metadataURI;
```

Comment:
extract it as parseMetadataURI method

Reply:
replaced by helper function in d6bd73e951951d08b925ccfe8c6a7cb4c91861f6